src/OndaEDF.jl

```diff
@@ -1,26 +1,131 @@
 module OndaEDF
 
 using Base.Iterators
-using Compat: @compat
-using Dates, UUIDs
-using Onda, EDF
+using Dates
+using EDF
+using Legolas
+using Onda
+using PrettyTables
 using StatsBase
 using TimeSpans
 using Tables
+using UUIDs
+
+using Compat: @compat
+using Legolas: @row, lift
+using Onda: LPCM_SAMPLE_TYPE_UNION, onda_sample_type_from_julia_type, convert_number_to_lpcm_sample_type
 using Tables: rowmerge
 
+export write_plan
+export edf_to_onda_samples, edf_to_onda_annotations, plan_edf_to_onda_samples, store_edf_as_onda
+export onda_to_edf
+
 # can be dropped if we drop Onda<0.14
 sample_type(x) = isdefined(Onda, :sample_type) ? Onda.sample_type(x) : x.sample_type
 
 include("standards.jl")
 
+"""
+    Plan = @row("ondaedf-plan@1",
+                # EDF.SignalHeader fields
+                label::String
+                transducer_type::String
+                physical_dimension::String
+                physical_minimum::Float32
+                physical_maximum::Float32
+                digital_minimum::Float32
+                digital_maximum::Float32
+                prefilter::AbstractString
+                samples_per_record::Int16
+                # EDF.FileHeader field,
+                seconds_per_record::Float64
+                # Onda.SamplesInfo fields (channels -> channel), may be missing
+                kind::Union{Missing, AbstractString}
+                channel::Union{Missing, AbstractString}
+                sample_unit::Union{Missing, AbstractString}
+                sample_resolution_in_unit::Union{Missing, LPCM_SAMPLE_TYPE_UNION}
+                sample_offset_in_unit::Union{Missing, LPCM_SAMPLE_TYPE_UNION}
+                sample_type::Union{Missing, AbstractString}
+                sample_rate::Union{Missing, LPCM_SAMPLE_TYPE_UNION}
+                # errors, use `nothing` to indicate no error
+                error::Union{Nothing, Exception}
+                )
+
+A type-alias for a Legolas row describing a single EDF signal-to-Onda channel
+conversion.  The columns are the union of
+- fields from `EDF.SignalHeader` (all mandatory)
+- the `seconds_per_record` field from `EDF.FileHeader` (mandatory)
+- fields from `Onda.SamplesInfo` (optional, may be `missing` to indicate failed
+  conversion)
+- `error`, which is `nothing` for a conversion that is or is expected to be 
+  successful, and the caught `Exception` for any failures.
+
+The [`FilePlan`](@ref) extension contiains two columns that give additional
+context about the conversion that only apply at the level of an `EDF.File`: 
+- `edf_signal_idx` gives the index of the source EDF signal
+- `onda_signal_idx` gives the index of the output `Onda.Samples`.  Note some
+  indices may be skipped in the output, so this is only to indicate order and
+  grouping.
+
+"""
+const Plan = @row("ondaedf-plan@1",
+                  # EDF.SignalHeader fields
+                  label::String = convert(String, label),
```

Comment:
I honestly do not know.  I guess it's conceivable that some other kidn of abstractstring comes in here that needs conversion, or (more likely) some of the fields that are `Float32` come in as other types, but I'm not sure.  I guess it can't hurt?

Reply:
`convert`s already happen automatically on construction of the row object:
```
julia> using Legolas

julia> struct Meme end

julia> Base.convert(::Type{Meme}, x::Int) = (println("gottem"); Meme())

julia> Bean = Legolas.@row("bean@1", meme::Meme)
Legolas.Row{Legolas.Schema{:bean, 1}, F} where F

julia> Bean((; meme=1))
gottem
Row(Schema("bean@1"), (meme = Meme(),))
```

Reply:
huh, TIL.  these are all based on what Onda.jl does.  I'll take em out anywy tho